src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -1087,23 +1093,31 @@ private ResultSet executeSPFkeys(String[] procParams) throws SQLException, SQLTi
             ResultSet userRs = null;
             PreparedStatement pstmt = null;
             StringBuilder azureDwSelectBuilder = new StringBuilder();
-            synchronized (SQLServerDatabaseMetaData.class) {
-                if (null == getImportedKeysDWColumns) {
-                    getImportedKeysDWColumns = new LinkedHashMap<>();
-                    getImportedKeysDWColumns.put(1, PKTABLE_CAT);
-                    getImportedKeysDWColumns.put(2, PKTABLE_SCHEM);
-                    getImportedKeysDWColumns.put(3, PKTABLE_NAME);
-                    getImportedKeysDWColumns.put(4, PKCOLUMN_NAME);
-                    getImportedKeysDWColumns.put(5, FKTABLE_CAT);
-                    getImportedKeysDWColumns.put(6, FKTABLE_SCHEM);
-                    getImportedKeysDWColumns.put(7, FKTABLE_NAME);
-                    getImportedKeysDWColumns.put(8, FKCOLUMN_NAME);
-                    getImportedKeysDWColumns.put(9, KEY_SEQ);
-                    getImportedKeysDWColumns.put(10, UPDATE_RULE);
-                    getImportedKeysDWColumns.put(11, DELETE_RULE);
-                    getImportedKeysDWColumns.put(12, FK_NAME);
-                    getImportedKeysDWColumns.put(13, PK_NAME);
-                    getImportedKeysDWColumns.put(14, DEFERRABILITY);
+
+            LinkedHashMap<Integer, String> importedKeysDWColumns = getImportedKeysDWColumns;
+            if (null == importedKeysDWColumns) {
+                LOCK.lock();
+                try {
+                    importedKeysDWColumns = getImportedKeysDWColumns;
```

Comment:
Bit confused about what's happening here. We grabbed the value, entered the block if its null, and grab it again. Is it possible that the value has changed by this point? If not, then I don't see the point of 1101 and 1102. Why isn't something like this possible?

`LOCK.lock()`
`try {`
`    --Lines 1091 - 1106--`
`} finally {`
`    LOCK.unlock()`
`}`

Like you have for the other cases.

Reply:
You are correct.  This is an error.  I'll commit a change for this.

Reply:
Well, it's not technically incorrect, but this is using double-checked locking (using a volatile).  Perhaps you don't like the idiom?